pyne/source_sampling.pyx

```diff
@@ -478,20 +478,65 @@ cdef class Sampler:
             rands_proxy = cpp_vector[double](<size_t> rands_size)
             for irands in range(rands_size):
                 rands_proxy[irands] = <double> rands[irands]
-        rtnval = (<cpp_source_sampling.Sampler *> self._inst).particle_birth(rands_proxy)
-        rtnval_proxy_shape[0] = <np.npy_intp> rtnval.size()
-        rtnval_proxy = np.PyArray_SimpleNewFromData(1, rtnval_proxy_shape, np.NPY_FLOAT64, &rtnval[0])
-        rtnval_proxy = np.PyArray_Copy(rtnval_proxy)
-        return rtnval_proxy
-    
-    
-    
+        cdef cpp_source_sampling.SourceParticle c_src = \
+                (<cpp_source_sampling.Sampler *> self._inst)\
+                .particle_birth(rands_proxy)
+        return SourceParticle(c_src.get_x(), c_src.get_y(), c_src.get_z(), \
+                c_src.get_e(), c_src.get_w(), c_src.get_c())
 
-    pass
 
 
+cdef class SourceParticle:
+    """Constructor for class SourceParticle
+    
+    Attributes
+    ----------
+    x (double) : The x coordinate of the source particle
+    y (double) : The y coordinate of the source particle
+    z (double) : The z coordinate of the source particle
+    e (double) : The energy of the source particle
+    w (double) : The weight of the source particle
+    c (int) : The cell number of the source particle
+    
+    Methods
+    -------
+    SourceParticle
+    ~SourceParticle
+    
+    Notes
+    -----
+    This class was defined in source_sampling.h
+    
+    The class is found in the "pyne" namespace"""
 
 
 
+    # constuctors
+    def __cinit__(self, double x, double y, double z, double e, double w, int c):
+        self.c_src = cpp_source_sampling.SourceParticle(x, y, z, e, w, c)
+    
+    def get_xyzew(self):
+        """get_xyzew(self)
+        Get source particle x, y, z, e and w
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        res1 : std::vector< double >
+        
+        """
+        cdef cpp_vector[double] rtnval
+        cdef np.npy_intp rtnval_proxy_shape[1]
+        rtnval = self.c_src.get_xyzew()
+        rtnval_proxy_shape[0] = <np.npy_intp> rtnval.size()
+        rtnval_proxy = np.PyArray_SimpleNewFromData(1, rtnval_proxy_shape, np.NPY_FLOAT64, &rtnval[0])
+        rtnval_proxy = np.PyArray_Copy(rtnval_proxy)
+        return rtnval_proxy
 
-{'cpppxd_footer': '', 'pyx_header': '', 'pxd_header': '', 'pxd_footer': '', 'cpppxd_header': '', 'pyx_footer': ''}
+    @property
+    def c(self):
+        return self.c_src.get_c()
```

Comment:
Don't you also want properties for x, y, z, w, e?

Reply:
I find that add properties for x, y, z, w and e will make test functions more readable. I add these properties and modified test functions.